llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -2287,6 +2302,13 @@ static bool eliminateDeadStores(Function &F, AliasAnalysis &AA, MemorySSA &MSSA,
 
   MadeChange |= State.eliminateRedundantStoresOfExistingValues();
   MadeChange |= State.eliminateDeadWritesAtEndOfFunction();
+
+  while (!State.ToRemove.empty()) {
+    Instruction *DeadInst = State.ToRemove.pop_back_val();
+    assert(!MSSA.getMemoryAccess(DeadInst));
```

Comment:
Nit: Perhaps an assertion comment along the lines of  "Dead MemDefs are erased right away"?

Reply:
I went ahead and dropped the assert, as it was incorrect and never triggered, due to memory accesses always being removed from MemorySSA.

Also, in the latest version, we may remove memory-defs here, if they produce values.